### PR TITLE
Set-up front-end

### PR DESCRIPTION
**Pull request by Alejandroq12**
Hello, dear reviewer.

- I made sure to complete the following requirements:
  - Create a new React app;
  - Initialize the project with Git;
  - Create the `App` component with react-router.
  - Create the `Greeting` component that will display a greeting. Set it up as a route in the App component.
  - Create a store, an action and a reducer that will connect to the API endpont to get the random greeting.
  - Display the random greeting in the `Greeting` component.
Screenshot from 2024-03-09 01-09-49

Thank you for your feedback.

### CODE REVIEW

File: src/slices/greetingSlice.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import axios from 'axios';
+
+const initialState = {
+  entity: '',
```
Comment (Omar-Muhamad) [inline src/slices/greetingSlice.js:5]: - [ ] Great job creating the redux slice here, Unfortunately, You need to add this slice inside the `redux` folder to maintain a professional file structure.

[Hint]
 ```
 - redux
    -- store.js
    -- greeting   // folder that includes the slice inside the redux folder
       --- greetingSlice.js
 ```

**Review by Omar-Muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @Alejandroq12,

### Nice job, you are almost there.

cat-driving-serious

## (Highlights) Good Points: :+1:
- Your work is well documented. ✔
- The correct workflow is used.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [x] The redux actions are firing twice and that's happening because of the strict mode that wraps your app in the `index.js` file. Please remove the strict mode because all actions should fire once only.

image

https://github.com/Alejandroq12/rails-react-frontend-two-app-template/blob/8d194afc99c9ce551f5ccdc6ff4b7577167b2b32/src/index.js#L11-L15

- [x] Please check the inline comments below ⬇️.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: src/slices/greetingSlice.js (hunk 1 above)
Comment (Alejandroq12) [reply src/slices/greetingSlice.js:5]: Thank you for your feedback.

**Review by Omar-Muhamad — APPROVED** [review]
# Approved ✔
### Hi @Alejandroq12,

#### Great job, Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

### Congratulations! 🎉 

giphy (1)

## (Highlights) Good Points: :+1:
- Your work is well documented. ✔
- The correct workflow is used.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- N/A

## [Optional] suggestions:

- N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**